api/bundle/client.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package bundle provides access to the bundle api facade.
+// This facade contains api calls that are specific to bundles.
+
+package bundle
+
+import (
+	"github.com/juju/loggo"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/apiserver/params"
+)
+
+var logger = loggo.GetLogger("juju.api.application")
+
+// Client allows access to the application API end point.
+type Client struct {
+	base.ClientFacade
+	st     base.APICallCloser
+	facade base.FacadeCaller
+}
+
+// NewClient creates a new client for accessing the bundle api.
+func NewClient(st base.APICallCloser) *Client {
```

Comment:
I was trying to work out for a bit why we didn't have one of these before. Then it became apparent that the facade method was only ever exposed for the GUI to call in to get the changes for the bundle deploy.